src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static Random random;
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(Random random, ColorSupplier colorSupplier) {
+        this.random = random;
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        String[] figureTypes = {"Square", "Rectangle",
+                "Right Rectangle", "Circle", "Isosceles Trapezoid"};
```

Comment:
Let's instead create a `FIGURES_COUNT` constant, where we would store value 5, and then we will generate a random number from 0 to 5 and check it in case statements. See next comment.